soco/core.py

```diff
@@ -1578,6 +1580,246 @@ def _build_album_art_full_uri(self, url):
     def _music_lib_search(self, search, start, max_items):
         return self.music_library._music_lib_search(search, start, max_items)
 
+    @only_on_master
+    def reorder_sonos_playlist(self, sonos_playlist, tracks, new_pos,
+                               update_id=0):
+        """Reorder and/or Remove tracks in a Sonos playlist.
+
+        The underlying call is quite complex as it both can move a track within
+        the list or delete a track from the playlist.  All of this depends on
+        what tracks and new_pos specify.
+
+        If a list is specified for tracks, then a list must be used for
+        new_pos. Each list element is a discrete modification and the next
+        list operation must anticipate the new state of the playlist.
+
+        If a comma formatted string to tracks is specified, then use
+        a similiar string to specify new_pos. Those operations should be
```

Comment:
Are you asking that I include an example?  i.e.
To reverse the order of the first 3 tracks, you should use  "2,0", "0,2"
 as compared to "0,2", "2,0".


On Mon, Sep 14, 2015 at 7:27 AM, Kenneth Nielsen <notifications@github.com>
wrote:

> In soco/core.py
> <https://github.com/SoCo/SoCo/pull/352#discussion_r39387395>:
>
> > @@ -1578,6 +1580,246 @@ def _build_album_art_full_uri(self, url):
> >      def _music_lib_search(self, search, start, max_items):
> >          return self.music_library._music_lib_search(search, start, max_items)
> >
> > +    @only_on_master
> > +    def reorder_sonos_playlist(self, sonos_playlist, tracks, new_pos,
> > +                               update_id=0):
> > +        """Reorder and/or Remove tracks in a Sonos playlist.
> > +
> > +        The underlying call is quite complex as it both can move a track within
> > +        the list or delete a track from the playlist.  All of this depends on
> > +        what tracks and new_pos specify.
> > +
> > +        If a list is specified for tracks, then a list must be used for
> > +        new_pos. Each list element is a discrete modification and the next
> > +        list operation must anticipate the new state of the playlist.
> > +
> > +        If a comma formatted string to tracks is specified, then use
> > +        a similiar string to specify new_pos. Those operations should be
>
> I'm a little unsure what you mean here by "Those operations should be
> ordered from the end of the list to the beginning"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SoCo/SoCo/pull/352/files#r39387395>.
>



-- 
Best,

Jeff Hinrichs
dundeemt@gmail.com
402.218.1473